### PR TITLE
162633506: Feature(User following) - Users should follow each other

### PR DESCRIPTION
**Pull request by fatukunda**
**What does this PR do?**

- Creates a user following feature

**Description of the tasks to be completed**

- Modify models, serializers, URLs, and views of the profile app to cater for the following functionality

- Modify the renderers file to include renderers for the **_following_** feature

- Add tests for this feature in the profiles tests folder

**How should this be manually tested?**

**Commands to run**
- `python manage.py makemigrations profiles`

- `python manage.py migrate profiles`

- `python manage.py runserver`

- `python manage.py test authors/apps/profiles/tests`

**Endpoints to be tested**

`POST /api/v1/users/:username/profile/follow`
`DELETE /api/v1/users/:username/profile/follow`
`GET / api/v1/users/me/profile/followers`
`GET / api/v1/users/me/profile/followings`

**What are the relevant PT stories?**

_**Delivers [#162633506](https://www.pivotaltracker.com/story/show/162633506)**_

### CODE REVIEW

**Comment by araaliFarooq** [comment]
LGTM

**Comment by NabunyaLilian** [comment]
LGTM

**Comment by mubstimor** [comment]
LGTM

**Comment by jeanjoe** [comment]
LGTM

File: authors/apps/profiles/tests/test_user_following.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    def test_registered_user_can_follow_another_registered_user(self):
+        user_1 = self.register_and_loginUser()
+        user_1_token = user_1.data['token']
+
+        user_2 = self.register_and_login_user2()
+        this_user_2 = User.objects.get(email=user_2.data['email'])
+
+        response = self.client.post(
+            '/api/v1/users/{}/profile/follow'.format(this_user_2.username), HTTP_AUTHORIZATION=f'Bearer {user_1_token}')
```
Comment (jeanjoe) [inline authors/apps/profiles/tests/test_user_following.py:20]: Make this line shorter than 80 Characters

File: authors/apps/profiles/tests/test_user_following.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    def test_a_registered_user_can_unfollow_a_user_they_follow(self):
+        user_1 = self.register_and_loginUser()
+        user_1_token = user_1.data['token']
+
+        user_2 = self.register_and_login_user2()
+        this_user_2 = User.objects.get(email=user_2.data['email'])
+        this_user_2.following = True
+
+        response = self.client.delete(
+            '/api/v1/users/{}/profile/follow'.format(this_user_2.username), HTTP_AUTHORIZATION=f'Bearer {user_1_token}')
```
Comment (jeanjoe) [inline authors/apps/profiles/tests/test_user_following.py:36]: Same here

File: authors/apps/profiles/tests/test_user_following.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                'profile']['following'], False
+        )
+
+    def test_a_user_cannot_follow_themselves(self):
+        user = self.register_and_loginUser()
+        user_token = user.data['token']
+        this_user = User.objects.get(email=user.data['email'])
+
+        response = self.client.post(
+            '/api/v1/users/{}/profile/follow'.format(this_user.username), HTTP_AUTHORIZATION=f'Bearer {user_token}')
```
Comment (jeanjoe) [inline authors/apps/profiles/tests/test_user_following.py:49]: Same here

**Review by jeanjoe — CHANGES_REQUESTED** [review]

**Review by jeanjoe — APPROVED** [review]

**Pull request closed without merging**